### PR TITLE
39 modal confirmation component

### PR DESCRIPTION
**Pull request by bochka123**

### CODE REVIEW

**Comment by bochka123** [comment]
image
image

File: frontend/src/app/shared/components/confirm/confirm.component.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<mat-dialog-content><div>{{data.header}}</div></mat-dialog-content>
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/shared/components/confirm/confirm.component.html:1]: Can you also add optional content?
Like header "Are you sure?"
Content: "You are about to delete your user, are you sure you want to do it"

File: frontend/src/app/shared/models/IModal.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export interface IModal {
+    header: string,
+    cancelText?: string,
+    confirmText?: string
+    okText?: string
+    /* eslint-disable  @typescript-eslint/no-explicit-any */
+    component?: any
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/shared/models/IModal.ts:7]: Can you use some sort of component class?
Or make something like this https://stackoverflow.com/questions/43933783/pass-component-to-a-component-in-angular

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

File: frontend/src/app/shared/models/IModal.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Type } from '@angular/core';
+
+export interface IModal {
+    header?: string,
+    content?: string,
+    cancelText?: string,
+    confirmText?: string
+    okText?: string
+    /* eslint-disable  @typescript-eslint/no-explicit-any */
+    component?: Type<any>
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/shared/models/IModal.ts:10]: I still don't like this any, but let it be then

**Review by Anatolii-Ivanov — APPROVED** [review]

File: frontend/src/app/shared/models/IModal.ts (hunk 3 above)
Comment (bochka123) [reply frontend/src/app/shared/models/IModal.ts:10]: I don`t like it too, but in Angular library *ngComponentOutlet is taking type Type<any> and if we take any other - there will be an error
image

**Review by ShevchenkoKirilo — APPROVED** [review]

**Pull request merged**